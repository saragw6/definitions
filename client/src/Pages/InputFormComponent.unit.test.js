import React from 'react';
import renderer from "react-test-renderer";
import InputForm from './InputFormComponent';

test('termRequest?', () => {
  const component = renderer.create(<InputForm labelInput="What term do you want defined?"/>);
  const jsonForm = component.toJSON();
  expect(jsonForm).toMatchSnapshot();

  const componentRequired = renderer.create(<InputForm labelInput="What term do you want defined?" isRequired={true}/>);
  const jsonFormRequired = componentRequired.toJSON();
  expect(jsonFormRequired).toMatchSnapshot();
});
