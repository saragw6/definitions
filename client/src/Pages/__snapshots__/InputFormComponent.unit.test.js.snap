exports[`test termRequest? 1`] = `
<div
  className="formInput">
  <label>
    What term do you want defined?
  </label>
   
  <div
    className="queerInput"
    data-react-toolbox="input">
    <input
      className=""
      data-testid="request-input"
      defaultValue={undefined}
      disabled={false}
      maxLength={50}
      name={undefined}
      onBlur={undefined}
      onChange={[Function]}
      onKeyPress={undefined}
      required={false}
      role="input"
      style={
        Object {
          "background": "transparent",
        }
      }
      type="Text"
      value={undefined} />
    <span
      className={undefined} />
    <span
      className={undefined}
      hidden={undefined}>
      Your answer
    </span>
    <span
      className={undefined}>
      0
      /
      50
    </span>
  </div>
</div>
`;

exports[`test termRequest? 2`] = `
<div
  className="formInput">
  <label>
    What term do you want defined?
  </label>
   
  <span
    className="error">
    *
  </span>
  <div
    className="queerInput"
    data-react-toolbox="input">
    <input
      className=""
      data-testid="request-input"
      defaultValue={undefined}
      disabled={false}
      maxLength={50}
      name={undefined}
      onBlur={undefined}
      onChange={[Function]}
      onKeyPress={undefined}
      required={false}
      role="input"
      style={
        Object {
          "background": "transparent",
        }
      }
      type="Text"
      value={undefined} />
    <span
      className={undefined} />
    <span
      className={undefined}
      hidden={undefined}>
      Your answer
    </span>
    <span
      className={undefined}>
      0
      /
      50
    </span>
  </div>
</div>
`;
